app/scripts/migrations/028.js

```diff
@@ -0,0 +1,37 @@
+// next version number
+const version = 28
+
+/*
+
+normalizes txParams on unconfirmed txs
+
+*/
+const clone = require('clone')
+
+module.exports = {
+  version,
+
+  migrate: async function (originalVersionedData) {
+    const versionedData = clone(originalVersionedData)
+    versionedData.meta.version = version
+    const state = versionedData.data
+    const newState = transformState(state)
+    versionedData.data = newState
+    return versionedData
+  },
+}
+
+function transformState (state) {
+  const newState = state
+
+  if (newState.PreferencesController) {
+    if (newState.PreferencesController.tokens) {
+      const tokens = newState.TransactionController.tokens
+      const selectedAddress = newState.PreferencesController.selectedAddress
+      newState.PreferencesController.tokens = []
+      newState.PreferencesController.accountTokens = {[selectedAddress]: {'mainnet': tokens}}
```

Comment:
So this migration removes tokens from all accounts except the main address. I think it would be more cautious/considerate to just keep all their tokens on all their accounts, so they don't notice any abrupt change.

This is so far the only thing I'd briefly block over.

Reply:
Sounds good, then users can remove/add tokens as they want.